### PR TITLE
3.6.2 fix android orientation crash

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by PatriceJiang** [comment]
Crash stack trace:

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 0 >>> com.dlprincess.guanfang.large <<<

backtrace:
  #00  pc 0x000000000121b860  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (v8::internal::Runtime::SetObjectProperty(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::StoreOrigin, v8::Maybe<v8::internal::ShouldThrow>)+24)
  #01  pc 0x0000000000ee806c  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)+240)
  #02  pc 0x0000000000aee84c  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (se::Object::setProperty(char const*, se::Value const&)+140)
  #03  pc 0x0000000000ab913c  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (cc::EventDispatcher::dispatchOrientationChangeEvent(int)+184)
  #04  pc 0x00000000000d7070  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/oat/arm64/base.odex (art_jni_trampoline+96)
  #05  pc 0x0000000000209188  /apex/com.android.art/lib64/libart.so (nterp_helper+152)
  #06  pc 0x0000000000813fb4  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/oat/arm64/base.vdex (com.cocos.lib.CocosOrientationHelper.access$100)
  #07  pc 0x0000000000209124  /apex/com.android.art/lib64/libart.so (nterp_helper+52)
  #08  pc 0x0000000000813f9e  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/oat/arm64/base.vdex (com.cocos.lib.CocosOrientationHelper$1.run+12)
  #09  pc 0x00000000002437fc  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/oat/arm64/base.odex (com.cocos.lib.CocosHelper$LockedTaskQ.runTasks+364)
  #10  pc 0x00000000002438e0  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/oat/arm64/base.odex (com.cocos.lib.CocosHelper.flushTasksOnGameThreadAtForeground+64)
  #11  pc 0x00000000002ca9e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568)
  #12  pc 0x00000000005ccbd0  /apex/com.android.art/lib64/libart.so (art::JNI<false>::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+644)
  #13  pc 0x0000000000897104  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+116)
  #14  pc 0x0000000000898460  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (void cc::JniHelper::callStaticVoidMethod<>(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&)+272)
  #15  pc 0x0000000000897f24  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so
  #16  pc 0x0000000000898618  /data/app/~~9u5Os-FCsmI8uGyYyQQh5Q==/com.dlprincess.guanfang.large-B_gGlmvhvR5SHP7jyrI_zg==/lib/arm64/libcocos.so (void* std::__ndk1::__thread_proxy<std::__ndk1::tuple<std::__ndk1::unique_ptr<std::__ndk1::__thread_struct, std::__ndk1::default_delete<std::__ndk1::__thread_struct> >, void (*)()> >(void*)+44)
  #17  pc 0x00000000000b6e94  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+264)
  #18  pc 0x00000000000534b0  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+68)
```

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

File: native/cocos/bindings/event/EventDispatcher.cpp [hunk 1]
```diff
@@ -39,8 +40,17 @@ se::Object *jsMouseEventObj = nullptr;
 se::Object *jsKeyboardEventObj = nullptr;
 se::Object *jsControllerEventArray = nullptr;
 se::Object *jsResizeEventObj = nullptr;
-se::Object *jsOrientationEventObj = nullptr;
 bool inited = false;
+
+// attach the argument object to the function
+void accessCacheArgObj(se::Object *func, se::Value *argObj, const char *cacheKey = "__reusedArgumentObject") {
```
Comment (bofeng-song) [inline native/cocos/bindings/event/EventDispatcher.cpp:46]: the argument name of orientation is __reusedArgumentObject ?

Comment (PatriceJiang) [reply native/cocos/bindings/event/EventDispatcher.cpp:46]: It is the field name assigned to the function.

**Review by bofeng-song — APPROVED** [review]

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Pull request merged**